### PR TITLE
Set default phase list for problems with multiple genuine entities

### PR DESCRIPTION
**Pull request by Christopher-Chianelli**
Before, all problems get the phase list [ConstructionHeuristic, LocalSearch] if
the phase list was not configured in the SolverConfig. This poses an issue for
problems with multiple genuine entities:
AbstractFromConfigFactory.deduceEntityDescriptor (which is used for
ConstructionHeuristic with a null entity placer config) throws an
exception if there are multiple genuine planning entities. As a result,
if you had a problem with multiple genuine planning entities, you are
forced to specify the phase list manually.

Now, all problems get the phase list
[
    ConstructionHeuristic(defaultEntityPlacer(genuineEntity1)),
    ConstructionHeuristic(defaultEntityPlacer(genuineEntity2)),
    ...
    ConstructionHeuristic(defaultEntityPlacer(genuineEntityN)),
    LocalSearch
]
if the phase list was not configured in the SolverConfig. Since
the EntityPlacer is configured with a specific genuine entity,
no exception is thrown.  Thus, phases no longer are required to
be configured for problems with multiple genuine entities.

Additionally, deduplicated code used to generate the default
entity placer (was duplicated between QueuedEntityPlacer and
PooledEntityPlacer).

Finally, fixed a bug in the MeetingSchedulingConstraintProvider,
where it used forEachIncludingNullVars but has a join that
depended on all variables (causing a rare NPE in construction
heuristic).

### Checklist
- [ ] Documentation updated if applicable.
- [ ] Upgrade recipe provided if applicable.

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* for a <b>pull request build</b> please add comment: <b>Jenkins retest this</b>
* for a <b>specific pull request build</b> please add comment: <b>Jenkins (re)run [optaplanner|kogito-apps|kogito-examples|optaplanner-quickstarts|optaweb-employee-rostering|optaweb-vehicle-routing] tests</b>
* for a <b>full downstream build</b> 
  * for <b>jenkins</b> job: please add comment: <b>Jenkins run fdb</b>
  * for <b>github actions</b> job: add the label `run_fdb`
* for a <b>compile downstream build</b> please add comment: <b>Jenkins run cdb</b>
* for a <b>full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>
* for an <b>upstream build</b> please add comment: <b>Jenkins run upstream</b>
* for a <b>Quarkus LTS check</b> please add comment: <b>Jenkins run LTS</b>
* for a <b>specific Quarkus LTS check</b> please add comment: <b>Jenkins (re)run [optaplanner|kogito-apps|kogito-examples|optaplanner-quickstarts|optaweb-employee-rostering|optaweb-vehicle-routing] LTS</b>
* for a <b>Native check</b> please add comment: <b>Jenkins run native</b>
* for a <b>specific Native LTS check</b> please add comment: <b>Jenkins (re)run [optaplanner|kogito-apps|kogito-examples|optaplanner-quickstarts|optaweb-employee-rostering|optaweb-vehicle-routing] native</b>
</details>

### CODE REVIEW

**Review by triceo — APPROVED** [review]
LGTM, but I'll leave the list variable code for @yurloc to review, as he is much better versed in that.

File: optaplanner-examples/src/main/java/org/optaplanner/examples/meetingscheduling/score/MeetingSchedulingConstraintProvider.java [hunk 1]
```diff
@@ -57,8 +57,7 @@
     // ************************************************************************
 
     protected Constraint roomConflict(ConstraintFactory constraintFactory) {
-        return constraintFactory.forEachIncludingNullVars(MeetingAssignment.class)
-                .filter(leftAssignment -> leftAssignment.getRoom() != null)
+        return constraintFactory.forEach(MeetingAssignment.class)
```
Comment (yurloc) [inline optaplanner-examples/src/main/java/org/optaplanner/examples/meetingscheduling/score/MeetingSchedulingConstraintProvider.java:60]: @Christopher-Chianelli is this anyhow related to the other changes in this PR?

Comment (Christopher-Chianelli) [reply optaplanner-examples/src/main/java/org/optaplanner/examples/meetingscheduling/score/MeetingSchedulingConstraintProvider.java:60]: Nope, but the `forEachIncludingNullVars` did catch a bug because it didn't set cache/selection order/sorter manner (where it cause a NPE).  If you look at the join, you see this:

```

                .join(MeetingAssignment.class,
                        equal(MeetingAssignment::getRoom),
                        lessThan(MeetingAssignment::getId),
                        overlapping(assignment -> assignment.getStartingTimeGrain().getGrainIndex(),
                                assignment -> assignment.getStartingTimeGrain().getGrainIndex() +
                                        assignment.getMeeting().getDurationInGrains()))
```

this requires room and startingTimeGrain to both not be null. Thus the `forEachIncludingNullVars` + room not null filter should be replaced by `forEach`.

**Review by yurloc — APPROVED** [review]
LGTM.

**Comment by yurloc** [comment]
CC @ge0ffrey please confirm this is OK if you have time to review this.
